File: Examples/Examples/Chart/Documentation Snippets/ChartDocsPie.cs
using System;
using TelerikUI;
using Foundation;
using System.Collections.Generic;

namespace Examples
{
	public class ChartDocsPie
	{
		TKChart chart = new TKChart();
		TKChartPieSeries series = null;

		public ChartDocsPie ()
		{
		}

		void snippet1 ()
		{
			// >> chart-pie-cs
			var pointsWithValueAndName = new List<TKChartDataPoint> ();
			pointsWithValueAndName.Add (new TKChartDataPoint (new NSNumber (20), null, "Google"));
			pointsWithValueAndName.Add(new TKChartDataPoint(new NSNumber(30), null, "Apple"));
			pointsWithValueAndName.Add (new TKChartDataPoint (new NSNumber (10), null, "Microsoft"));
			pointsWithValueAndName.Add (new TKChartDataPoint (new NSNumber (5), null, "IBM"));
			pointsWithValueAndName.Add (new TKChartDataPoint (new NSNumber (8), null, "Oracle"));

			var series = new TKChartPieSeries (pointsWithValueAndName.ToArray());
			series.Style.PointLabelStyle.TextHidden = false;

			chart.AddSeries (series);
			chart.Legend.Hidden = false;
			chart.Legend.Style.Position = TKChartLegendPosition.Right;
			// << chart-pie-cs
		}

		void snippet2 ()
		{
			// >> chart-pie-format-cs
			series.LabelDisplayMode = TKChartPieSeriesLabelDisplayMode.Inside;

			var numberFormatter = new NSNumberFormatter ();
			numberFormatter.NumberStyle = NSNumberFormatterStyle.SpellOut;
			series.Style.PointLabelStyle.Formatter = numberFormatter;
			// << chart-pie-format-cs
		}

		void snippet3()
		{
			// >> chart-pie-format-percent-cs
			series.Style.PointLabelStyle.StringFormat = "%.0f %%";
			// << chart-pie-format-percent-cs
		}

		void snippet4()
		{
			// >> chart-pie-angle-cs
			series.StartAngle = (float)(-Math.PI/4.0/2.0);
			series.EndAngle = (float)(Math.PI/4.0/2.0);
			series.RotationAngle = (float)Math.PI;
			// << chart-pie-angle-cs
		}

		void snippet5()
		{
			// >> chart-pie-select-cs
			series.Selection = TKChartSeriesSelection.DataPoint;
			chart.Select(new TKChartSelectionInfo(series, 1));
			// << chart-pie-select-cs
		}
	}
}

File: TelerikUIExamplesInSwift/TelerikUIExamplesInSwift/ChartDocsPie.swift
//
//  ChartDocsPie.swift
//  TelerikUIExamplesInSwift
//
//  Copyright © 2016 Telerik. All rights reserved.
//

class ChartDocsPie: UIViewController {
    var pointsWithValueAndName = [TKChartDataPoint]()
    var chart = TKChart()
    
    override func viewDidLoad() {
        super.viewDidLoad()
        chart.frame = CGRectInset(self.view.bounds, 10, 10)
        self.view.addSubview(chart)
        
    }
    
    func pieSeries() {
        // >> chart-pie-swift
        pointsWithValueAndName.append(TKChartDataPoint(name: "Google", value: 20))
        pointsWithValueAndName.append(TKChartDataPoint(name: "Apple", value: 30))
        pointsWithValueAndName.append(TKChartDataPoint(name: "Microsoft", value: 10))
        pointsWithValueAndName.append(TKChartDataPoint(name: "IBM", value: 5))
        pointsWithValueAndName.append(TKChartDataPoint(name: "Oracle", value: 8))
        
        let series = TKChartPieSeries(items: pointsWithValueAndName)
        chart.addSeries(series)
        chart.legend.hidden = false
        chart.legend.style.position = TKChartLegendPosition.Right
        // << chart-pie-swift
        
        // >> chart-pie-format-swift
        series.labelDisplayMode = TKChartPieSeriesLabelDisplayMode.Inside
        
        let  numberFormatter = NSNumberFormatter()
        numberFormatter.numberStyle = NSNumberFormatterStyle.SpellOutStyle
        series.style.pointLabelStyle.formatter = numberFormatter
        // << chart-pie-format-swift
        
        // >> chart-pie-format-percent-swift
        series.style.pointLabelStyle.stringFormat = "%.0f %%"
        // << chart-pie-format-percent-swift
    }
    
    func formatLabelsWithLabelFormatter() {
        let series = TKChartPieSeries(items: pointsWithValueAndName)
        chart.addSeries(series)
    }
    
    func startEndAngles() {
        let series = TKChartPieSeries(items: pointsWithValueAndName)
        chart.addSeries(series)
        
        // >> chart-pie-angle-swift
        series.startAngle = CGFloat(-M_PI_4 / 2)
        series.endAngle = CGFloat(M_PI + M_PI_4 / 2)
        series.rotationAngle = CGFloat(M_PI)
        // << chart-pie-angle-swift
    }
    
    func handleSelection() {
        let series = TKChartPieSeries(items: pointsWithValueAndName)
        chart.addSeries(series)
        
        // >> chart-pie-select-swift
        series.selection = TKChartSeriesSelection.DataPoint
        chart.select(TKChartSelectionInfo(series: series, dataPointIndex: 1))
        // << chart-pie-select-swift
    }

}
